### ISSUE TITLE
`Due Date` not populating on AMD Service Request 

### ISSUE THREAD

**Issue by dianamartin** [cityofaustin/atd-knack-data-tracker#240]
Jared Wall
AMD

Came by and noticed that a `Due Date` wasn't populating on one of his CSR service requests.

**Comment by dianamartin** [comment]
Did some initial troubleshooting with Jared. Checked the data publication log to see if script was running. There was an issue on 9/8 and 9/9. The SR was created on 9/8, so that might've been the issue. 

I ran a query on the back end and checked to see if there were any more `Due Dates` not populating and there were a lot that came back blank. 

@johnclary we need to look into this issue. Not sure if it was due to the stop of service for the server migration that Knack did or this was something else.

**Comment by johnclary** [comment]
@dianamartin thanks for that research. i put this on deck for sprint 12.

**Comment by johnclary** [comment]
@dianamartin would you check some recent SRs that are missing due dates and see if they have a "due date" flex note?

**Comment by dianamartin** [comment]
#### Spot Testing

`19-00438627`
created: 11/15/2019
shows due date: 11/3/2020
has flex note `Created Date` field

`19-00441042` - Close (Resolved)
created: 11/18/2019
does not show due date: `blank`
has flex note `Created Date` field

**Comment by dianamartin** [comment]
@johnclary I've spot checked and even checked a few from the date range that had a publication issue: 9/8/19 and 9/9/2019. It's weird, you can see my notes on what I checked.

**Comment by amenity** [comment]
Blocked by migration of Knack Proxy server.

**Comment by amenity** [comment]
@johnclary - can you link this up with the Knack Proxy issues that are blocking it, please?

**Comment by dianamartin** [comment]
John said that this issue of the `Due Date` field not populating was because the CSR Issue didn't have an associated Flex Note. Apparently, that's where the due date gets generated and plugged into the CSR. 

@johnclary bc of this reason, can we close this issue?

**Comment by johnclary** [comment]
No, it cannot be closed. We're waiting on Omar to update the ESB to point to new our new Lambda function on AWS. See this issue's dependency.

**Comment by dianamartin** [comment]
I thought it wasn't related to the knack server issue?

**Comment by johnclary** [comment]
It is. The on-premise proxy server is occasionally throwing a mysterious error, which we think is related to the server load being too high. The ESB is supposed to keep retrying to send messages when it receives an error, but for some reason the flex notes occasionally do not make it over to Knack.

Our hope is that the new knack proxy server will eliminate these errors because it is more robust.

**Comment by dianamartin** [comment]
Confirmed with John that this proxy server was migrated. I will need to check the signal CSRs to see if the `Due Date` field is being populated.

**Comment by dianamartin** [comment]
@johnclary I checked Data Tracker and this is what I noticed: 

`FEB YTD (2/6/2020)`
- 155 records
- 109 blank
- 46 due date field

`JAN`
- 1171 records
- 803 blank
- 368  due date field

I'm not sure if this is a normal amount that have the `Due_Date` field populated or not.

**Comment by johnclary** [comment]
@dianamartin i made a small change to the integration on Thursday that seems to have resolved the issues with missing flex notes. it won't fix the historical records, but this should be resolved going forward. keep an eye on it and let me know if otherwise. thanks.
